lib/chef/resource/windows_package.rb

```diff
@@ -69,10 +71,22 @@ def source(arg=nil)
           @source
         else
           raise ArgumentError, "Bad type for WindowsPackage resource, use a String" unless arg.is_a?(String)
-          Chef::Log.debug("#{package_name}: sanitizing source path '#{arg}'")
-          @source = ::File.absolute_path(arg).gsub(::File::SEPARATOR, ::File::ALT_SEPARATOR)
+          if uri_scheme?(arg)
+            @source = arg
+          else
+            @source = ::File.absolute_path(arg).gsub(::File::SEPARATOR, ::File::ALT_SEPARATOR)
```

Comment:
Probably should switch to ```Chef::Util::PathHelper.canonical_path``` now, which would also add long path support with the \\?\ business.